parser/Builder.cc

```diff
@@ -223,10 +223,13 @@ class Builder::Impl {
         return make_unique<Hash>(collectionLoc(begin, pairs, end), std::move(pairs));
     }
 
-    unique_ptr<Node> attrAsgn(unique_ptr<Node> receiver, const token *dot, const token *selector) {
+    unique_ptr<Node> attrAsgn(unique_ptr<Node> receiver, const token *dot, const token *selector, bool masgn) {
         core::NameRef method = gs_.enterNameUTF8(selector->string() + "=");
         core::LocOffsets loc = receiver->loc.join(tokLoc(selector));
         if ((dot != nullptr) && dot->string() == "&.") {
+            if (masgn) {
+                error(ruby_parser::dclass::CSendInLHSOfMAsgn, tokLoc(dot));
+            }
```

Comment:
I feel like it would have been cleaner if done in `parser.yyp` instead of the Builder but it seems that all the other error cases are always handled in the builder.

If done here, we need to know if the assignation is on a multi left hand side, hence the extra parameter.

Reply:
I kind of like your suggestion of inlining this branch into the two cases where `masgn` is `true`.

If you'd like to change this, let me know. I'll merge this at the end of the day or whenever you get back to me.